src/main/java/dev/jbang/ScriptResource.java

```diff
@@ -13,11 +15,19 @@
 	private File sourceCacheDir;
 	// physical file it is mapped to
 	private File file;
+	// This handles when main script file is in a package.
+	// eg: /home/user/scripts/org/acme/Main.java, then
+	// originalSourceFileBaseDir = /home/user/scripts
+	private final File originalSourceFileBaseDir;
 
 	public ScriptResource(String scriptURL, File urlCache, File file) {
 		this.originalResource = scriptURL;
 		this.sourceCacheDir = urlCache;
 		this.file = file;
+		if (Util.isJavaFile(scriptURL))
+			this.originalSourceFileBaseDir = setOriginalSourceFileBaseDir(scriptURL);
```

Comment:
this reads wrong - why call setOriginalSourceFileBaseDir(scriptUrl) to get the return value ...why is this not just calculated (and cached) in `getOriginalSourceFileBaseDir()` ?